client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/protocol/InternalFlag.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.client.hotrod.impl.protocol;
+
+/**
+ * Flags which are internal to the protocol and shouldn't be exposed to users.
```

Comment:
> @galderz we can move the lifespan and maxIdle values in the body, if that's your only concern :)

lifespan and maxIdle are already in the body. I just want to avoid as much as conditional decoding server side.

> On a more serious note, I would really hate to send two 64-bit numbers in each write request that doesn't need them. I would maybe be ok with two 32-bit numbers, but that would still require some extra logic to change the unit automatically, in order to avoid overflows.

For optional fields, it might make sense to have variable length numbers, there's certainly room for those, but some others, e.g. an Message ID, which is mandatory, might be better to have a 64 bit number and then read the entire header in one single go. The devil's in the detail :)